lib/common/common/Int64Math.cpp

```diff
@@ -30,6 +37,16 @@ Int64Math::Mul(int64 left, int64 right, int64 *pResult)
     return (left != 0 && right != 0 && (*pResult / left) != right);
 #endif
 }
+#else
+// FIXME: this probably needs checks for __int128 at least.
+bool
+Int64Math::Mul(int64 left, int64 right, int64 *pResult)
+{
+    __int128 result = (__int128)left * (__int128)right;
+    *pResult = (int64)result;
+    return (left != 0 && right != 0 && (*pResult / left) != right);
```

Comment:
Yep, I suggested something very similar in https://github.com/xanlpz/ChakraCore/commit/eb5d8912eec17a8132f21b714fd1daeed0d626a6 (but checking for _MSC_VER is probably better)